### PR TITLE
WIP - Create single XCFramework for all platforms

### PR DESCRIPTION
**Pull request by DavidBrunow**
This is based upon @blender's PR for adding support for XCFrameworks #2801 

Current known issues that need to be fixed:

- [ ] Current implementation only builds a single XCFramework with all combined platforms and does not currently support building frameworks split by platform
- [ ] Current implementation must have a derived data directory set in order to avoid issues with builds locking files on each other

I made a design decision to output the combined XCFramework to the `Carthage/Build` folder and would appreciate input on that decision.

### CODE REVIEW

**Comment by DavidBrunow** [comment]
FYI I am still working on this and making it work for my specific use case but won't have bandwidth to generalize the implementation to work for everyone until after iOS 13 comes out.

File: Source/carthage/Update.swift [hunk 1]
```diff
@@ -65,15 +65,18 @@ public struct UpdateCommand: CommandProtocol {
 			let buildDescription = "skip the building of dependencies after updating\n(ignored if --no-checkout option is present)"
 
 			let dependenciesUsage = "the dependency names to update, checkout and build"
+			let defaultValue: String? = nil
```
Comment (jdhealy) [inline Source/carthage/Update.swift]: @DavidBrunow — as you're probably already aware (but hopefully positively expecting), the merging of https://github.com/Carthage/Carthage/pull/2859 into `master` will cause merge conflicts with this hunk.

And a way over due: major thanks for working on this 🙂👍

**Comment by Aranoledur** [comment]
@DavidBrunow @tmspzz Hey, guys! Can we set up goals for this feature? Our company need this feature in release and ok for 1 person working on it fulltime

**Comment by tmspzz** [comment]
@Aranoledur let me take put something together

**Comment by DavidBrunow** [comment]
@tmspzz Who would make the decision that having the ability to create a single XCFramework for all platforms is something that Carthage should do? I want to make sure we have buy-in that that is valuable.

**Comment by tmspzz** [comment]
@DavidBrunow long story short, it's desirable to support both. Before you continue though, please allow me some more time to post a "roadmap" for what the goals are for Carthage in the near future. I need a run it by @Carthage/carthage and then will share it and gather feedback

**Comment by tmspzz** [comment]
@Aranoledur please see this: https://github.com/Carthage/Carthage/issues/2890

**Comment by DavidBrunow** [comment]
Still working on this...

**Pull request closed without merging**